File: EIDSS v5/android/EIDSS/HumanCaseListActivity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using eidssdroid.Model;
using eidssdroid.model.Database;

namespace EIDSS
{
    [Activity]
    public class HumanCaseListActivity : Activity
    {
        private ListView m_listView;
        private List<HumanCase> m_cases;

        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);

            SetTitle(Resource.String.TitleHumanCaseList);
            SetContentView(Resource.Layout.HumanCaseListLayout);
            Button buttonOk = FindViewById<Button>(Resource.Id.OkButton);
            Button buttonNewHumanCase = FindViewById<Button>(Resource.Id.NewHumanCaseButton);
            Button buttonSynchronizeHumanCase = FindViewById<Button>(Resource.Id.SynchronizeHumanCaseButton);
            Spinner spinnerCaseStatusFilter = FindViewById<Spinner>(Resource.Id.CaseStatusFilterLookup);
            
            var adapter = ArrayAdapter.CreateFromResource (this, Resource.Array.CaseStatusFilterItems, Android.Resource.Layout.SimpleSpinnerItem);
            adapter.SetDropDownViewResource (Android.Resource.Layout.SimpleSpinnerDropDownItem);
            spinnerCaseStatusFilter.Adapter = adapter;

            spinnerCaseStatusFilter.ItemSelected += CaseStatusFilter_spinner_ItemSelected;
            
            m_listView = FindViewById<ListView>(Resource.Id.List); 

            using (var mDb = new EidssDatabase(this))
            {
                m_cases = mDb.HumanCaseSelect();
                m_listView.Adapter = new HumanCaseListAdapter(this, m_cases);
            }

            m_listView.ItemLongClick += (sender, args) =>
            {
                var t = m_cases[args.Position];
                var i = new Intent(ApplicationContext, typeof(HumanCaseActivity));
                i = i.PutExtra("hc", t);
                StartActivityForResult(i, 333);
            };

            buttonNewHumanCase.Click += delegate
            {
                var i = new Intent(ApplicationContext, typeof(HumanCaseActivity));
                var mCase = HumanCase.CreateNew();
                i = i.PutExtra("hc", mCase);
                StartActivityForResult(i, 333);
            };

            buttonSynchronizeHumanCase.Click += delegate
            {
                var i = new Intent(ApplicationContext, typeof(SynchronizeHumanCaseActivity));
                StartActivityForResult(i, 444);
            };

            buttonOk.Click += delegate
            {
                this.Finish();
            };

        }

        void CaseStatusFilter_spinner_ItemSelected(object sender, AdapterView.ItemSelectedEventArgs e)
        {
            using (var mDb = new EidssDatabase(this))
            {
                List<HumanCase> cases = mDb.HumanCaseSelect();

                switch (e.Position)
                {
                    case 0:
                        m_cases = cases;
                        break;
                    case 1:
                        m_cases = cases.Where(c => c.intStatus == (int)HumanCaseStatus.NEW).ToList();
                        break;
                    case 2:
                        m_cases = cases.Where(c => c.intStatus == (int)HumanCaseStatus.CHANGED).ToList();
                        break;
                    case 3:
                        m_cases = cases.Where(c => c.intStatus == (int)HumanCaseStatus.SYNCHRONIZED).ToList();
                        break;
                }

                m_listView.Adapter = new HumanCaseListAdapter(this, m_cases);
            }
        }

        protected override void OnActivityResult(int requestCode, Result resultCode, Intent data)
        {
            if (requestCode == 333)
            {
                if (resultCode == Result.FirstUser) // delete
                {
                    var mCase = (HumanCase)data.GetParcelableExtra("hc");
                    using (var db = new EidssDatabase(this))
                    {
                        db.HumanCaseDelete(mCase);
                    }
                    this.Recreate();
                }
                if (resultCode == Result.Ok)
                {
                    var mCase = (HumanCase)data.GetParcelableExtra("hc");
                    using (var db = new EidssDatabase(this))
                    {
                        if (mCase.id == 0)
                        {
                            db.HumanCaseInsert(mCase);
                        }
                        else
                        {
                            if (mCase.idfCase != 0)
                                mCase.intStatus = (int)HumanCaseStatus.CHANGED;
                            db.HumanCaseUpdate(mCase);
                        }
                    }
                    this.Recreate();
                }
            }
            if (requestCode == 444)
            {
                if (resultCode == Result.Ok)
                {
                    this.Recreate();
                }
            }
            base.OnActivityResult(requestCode, resultCode, data);
        }
    }
}

File: EIDSS v5/android.java/workspace/EIDSS/src/com/bv/eidss/HumanCaseListActivity.java
package com.bv.eidss;

import java.util.List;

import com.bv.eidss.data.EidssDatabase;
import com.bv.eidss.model.HumanCase;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.AdapterView.OnItemSelectedListener;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Spinner;

public class HumanCaseListActivity extends Activity {

	private Activity _this;
	public HumanCaseListActivity() {
    	_this = this;
	}
	
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setTitle(R.string.TitleHumanCaseList);
        setContentView(R.layout.human_case_list_layout);
   	
        final ListView listView = (ListView)findViewById(R.id.List);
        
        EidssDatabase mDb = new EidssDatabase(this);
        List<HumanCase> cases = mDb.HumanCaseSelect(0);
        mDb.close();
        listView.setAdapter(new HumanCaseListAdapter(this, cases));
        
        listView.setOnItemLongClickListener(new ListView.OnItemLongClickListener() {
			@Override
			public boolean onItemLongClick(AdapterView<?> list, View arg1, int position, long arg3) {
				Intent intent = new Intent();
				intent = intent.setClass(getApplicationContext(), HumanCaseActivity.class);
				HumanCaseListAdapter adapter = (HumanCaseListAdapter)list.getAdapter();
				HumanCase hc = (HumanCase)adapter.getItem(position);
				intent = intent.putExtra("hc", hc);
				_this.startActivityForResult(intent, 333);
				return false;
			}
		});
    	findViewById(R.id.NewHumanCaseButton).setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				Intent intent = new Intent();
				intent = intent.setClass(getApplicationContext(), HumanCaseActivity.class);
				intent = intent.putExtra("hc", HumanCase.CreateNew());
				_this.startActivityForResult(intent, 333);
			}
		});
    	findViewById(R.id.SynchronizeHumanCaseButton).setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				Intent intent = new Intent();
				intent = intent.setClass(getApplicationContext(), SynchronizeHumanCaseActivity.class);
				_this.startActivityForResult(intent, 444);
			}
		});
    	findViewById(R.id.OkButton).setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				_this.finish();
			}
		});
     
        Spinner spinnerCaseStatusFilter = (Spinner)findViewById(R.id.CaseStatusFilterLookup);
        //ArrayAdapter<String> adapter = new ArrayAdapter<String>(this, R.array.CaseStatusFilterItems, android.R.layout.simple_spinner_item);
        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(this, R.array.CaseStatusFilterItems, android.R.layout.simple_spinner_item);
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        spinnerCaseStatusFilter.setAdapter(adapter);
        spinnerCaseStatusFilter.setOnItemSelectedListener(new OnItemSelectedListener(){
			@Override
			public void onItemSelected(AdapterView<?> parent, View v, int position, long id) {
		        EidssDatabase mDb = new EidssDatabase(_this);
		        List<HumanCase> cases = mDb.HumanCaseSelect(position);
		        mDb.close();
		        listView.setAdapter(new HumanCaseListAdapter(_this, cases));
			}
			@Override
			public void onNothingSelected(AdapterView<?> arg0) {
			}});
    }

	@Override
    public void onActivityResult(int requestCode, int resultCode, Intent data)
    {
        if (requestCode == 333)
        {
            if (resultCode == RESULT_FIRST_USER) // delete
            {
            	HumanCase mCase = (HumanCase)data.getParcelableExtra("hc");
            	EidssDatabase db = new EidssDatabase(this);
                db.HumanCaseDelete(mCase);
                db.close();
                //this.recreate();
                Intent intent = getIntent();
                finish();
                startActivity(intent);
            }
            if (resultCode == RESULT_OK)
            {
            	HumanCase mCase = (HumanCase)data.getParcelableExtra("hc");
            	EidssDatabase db = new EidssDatabase(this);
                if (mCase.getId() == 0)
                {
                    db.HumanCaseInsert(mCase);
                }
                else
                {
                    if (mCase.getCase() != 0)
                        mCase.setStatusChanged();
                    db.HumanCaseUpdate(mCase);
                }
                db.close();
                //this.recreate();
                Intent intent = getIntent();
                finish();
                startActivity(intent);
            }
        }
        if (requestCode == 444)
        {
            if (resultCode == RESULT_OK)
            {
                //this.recreate();
                Intent intent = getIntent();
                finish();
                startActivity(intent);
            }
        }

        super.onActivityResult(requestCode, resultCode, data);
    }
    
}
